src/org/labkey/targetedms/SkylineDocImporter.java

```diff
@@ -233,465 +234,472 @@ private void importSkylineDoc(TargetedMSRun run, File f) throws XMLStreamExcepti
+            _transitionChromInfoAnnotationStmt = null;
+            if (_transitionAnnotationStmt != null) { try { _transitionAnnotationStmt.close(); } catch (SQLException ignored) {} }
+            _transitionAnnotationStmt = null;
+            if (_precursorChromInfoAnnotationStmt != null) { try { _precursorChromInfoAnnotationStmt.close(); } catch (SQLException ignored) {} }
+            _precursorChromInfoAnnotationStmt = null;
+            if (_precursorAnnotationStmt != null) { try { _precursorAnnotationStmt.close(); } catch (SQLException ignored) {} }
+            _precursorAnnotationStmt = null;
+            if (_transitionChromInfoStmt != null) { try { _transitionChromInfoStmt.close(); } catch (SQLException ignored) {} }
+            _transitionChromInfoStmt = null;
+            if (_precursorChromInfoStmt != null) { try { _precursorChromInfoStmt.close(); } catch (SQLException ignored) {} }
+            _precursorChromInfoStmt = null;
+            if (_generalMoleculeAnnotationStmt != null) { try { _generalMoleculeAnnotationStmt.close(); } catch (SQLException ignored) {} }
+            _generalMoleculeAnnotationStmt = null;
+        }
+    }
+
+    private ReplicateInfo insertReplicates(TargetedMSRun run, SkylineDocumentParser parser, OptimizationSettings optimizationSettings, TargetedMSService.FolderType folderType)
+    {
+        ReplicateInfo replicateInfo = new ReplicateInfo();
+
+        Map<Instrument, Integer> instrumentIdMap = new HashMap<>();
+        for(SkylineReplicate skyReplicate: parser.getReplicates())
+        {
+            Replicate replicate = new Replicate();
+            replicate.setName(skyReplicate.getName());
+            replicate.setSampleFileList(skyReplicate.getSampleFileList());
+            replicate.setAnnotations(skyReplicate.getAnnotations());
+            replicate.setSampleType(skyReplicate.getSampleType());
+            replicate.setAnalyteConcentration(skyReplicate.getAnalyteConcentration());
+            replicate.setSampleDilutionFactor(skyReplicate.getSampleDilutionFactor());
+            replicate.setRunId(_runId);
+
+            if(optimizationSettings._cePredictor != null && skyReplicate.getCePredictor() != null && optimizationSettings._cePredictor.equals(skyReplicate.getCePredictor()))
+            {
+                replicate.setCePredictorId(optimizationSettings._cePredictor.getId());
+            }
+            if(optimizationSettings._dpPredictor != null && skyReplicate.getDpPredictor() != null && optimizationSettings._dpPredictor.equals(skyReplicate.getDpPredictor()))
+            {
+                replicate.setDpPredictorId(optimizationSettings._dpPredictor.getId());
+            }
 
-                    if (folderType == TargetedMSService.FolderType.QC)
+            if (folderType == TargetedMSService.FolderType.QC)
+            {
+                // In QC folders insert a replicate only if at least one of the associated sample files will be inserted
```

Comment:
This comment should be removed, I think.  We now keep replicates and sample files from the latest uploaded Skyline document and delete older ones.  This comment (and the one on line 1939) look like they are from before this change.

Reply:
I think those comments are still relevant. While we retain older data, we still need to insert newly added replicates and sample files.